build.gradle

```diff
@@ -411,6 +411,10 @@ def calculateVersion() {
 
 def getCheckedOutGitCommitHash() {
   def gitFolder = "$projectDir/.git/"
+  if (!file(gitFolder).isDirectory()) {
+    // we may be in a submodule
+    gitFolder = file(gitFolder).text.substring(8).trim() + "/"
```

Comment:
Might be worth adding an explanatory comment here

Reply:
Done